# Microsoft Film Cration Strategies Analysis

![example](https://github.com/learn-co-curriculum/dsc-project-template/blob/template-mvp/images/director_shot.jpeg?raw=1)

## Overview

[Microsoft](https://en.wikipedia.org/wiki/Microsoft) as one of the most renounded tech companies in the world, popular for developing the Windows Operating System, is looking to dive into the industry of film creation. This project aims to provide guiding insights to Microsoft as it begins its content creation journey by analyzing popular Box Office films from other large companies. This analysis mainly focuses on the genre, runtime, rating and the producing studio of films that perform best at Box Office.

## Business Problem

Microsoft is planning to launch a new movie studio but **lacks insight** into what types of films are most successful in today's market. By analyzing data on box office perfomance, genres, production studios and audience ratings, **we aim to identify trends that can guide Microsoft's content creation strategy and maximize both domestic and international success**.

## Data Understanding

The data files provide; the title of the film, year of creation of the film, the average rating of the film as well as other attributes of the film (e.g. film runtime, domestic and international profits).
Every film has a unique title ID associated with the stated attributes.

### Reading and Previewing data

In [131]:
import pandas as pd

In [132]:
box_office_data = pd.read_csv('bom.movie_gross.csv')
movie_info = pd.read_csv('title.basics.csv')
ratings_data = pd.read_csv('title.ratings.csv')

#### Box Office Film Data

In [133]:
box_office_data.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [134]:
box_office_data.shape

(3387, 5)

In [135]:
box_office_data.columns

Index(['title', 'studio', 'domestic_gross', 'foreign_gross', 'year'], dtype='object')

In [136]:
box_office_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


The box office information has a total of 3387 rows and 5 columns. There are categorical columns and numerical columns (with one having integers and another having floats).

There appears to be missing values in the studio, domestic_gross and foreign_gross columns.

#### Movie Information Data

In [137]:
movie_info.head()

tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [138]:
movie_info.columns

Index(['tconst', 'primary_title', 'original_title', 'start_year',
       'runtime_minutes', 'genres'],
      dtype='object')

In [139]:
movie_info.shape

(146144, 6)

In [140]:
movie_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   tconst           146144 non-null  object 
 1   primary_title    146143 non-null  object 
 2   original_title   146122 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


The movie information data has a total of 146144 rows and 6 columns. There are categorical columns and numerical columns (with one having integers and another having floats).


There appears to be missing values in majority of the columns.

#### Ratings Data

In [141]:
ratings_data.head()

tconst  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [142]:
ratings_data.shape

(73856, 3)

In [143]:
ratings_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   tconst         73856 non-null  object 
 1   averagerating  73856 non-null  float64
 2   numvotes       73856 non-null  int64  
dtypes: float64(1), int64(1), object(1)
memory usage: 1.7+ MB


The ratings information data has a total of 73856 rows and 3 columns. Each column has a different datatype (string, float and integer resp.) and there are no missing values in either of the columns.

## Data Cleaning

### Box Office Information Data

For the box office information data, there was a small percentage of missing values in the studio column and domestic_ gross column which could be solved by dropping the rows.

A large fraction of the foreign gross column was missing as well. I decided that it would be best to drop the rows instead of filling them with the mean/median as it would sway the analysis which would yield erenous insights.

I removed comma separators from values in the foreign_gross column changed its datatype to float so that I can use it for calculation.



In [144]:
box_office_data.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [145]:
box_office_data = box_office_data.dropna(subset=['studio', 'domestic_gross', 'foreign_gross'])
box_office_data.isna().sum()

title             0
studio            0
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

In [ ]:
box_office_data['foreign_gross'] = box_office_data['foreign_gross'].str.replace(',', '')
box_office_data['foreign_gross'] = box_office_data['foreign_gross'].astype(float)

### Movie Information Data

For the movie information data, I did the same procedure as the box office information data which was to drop the missing data as I lacked the required information to fill in the missing values.

I then separated the genres into individual strings and extracted the first genre and added it to a new column called Main Genre for grouping later on.

In [147]:
movie_info.isna().sum()

tconst                 0
primary_title          1
original_title        22
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64

In [148]:
movie_info = movie_info.dropna(subset=['primary_title', 'original_title', 'runtime_minutes', 'genres'])
movie_info.isna().sum()

tconst             0
primary_title      0
original_title     0
start_year         0
runtime_minutes    0
genres             0
dtype: int64

In [ ]:
movie_info['Main_genres'] = movie_info['genres'].map(lambda x: x.split(',')[0])

In [150]:
movie_info['Main_genres'].value_counts()

Main_genres
Documentary    33766
Drama          24535
Comedy         17481
Action          8110
Biography       7564
Horror          4974
Adventure       4243
Crime           3315
Thriller        1931
Animation       1430
Romance          884
Family           774
Mystery          652
Fantasy          646
Sci-Fi           535
Music            436
History          266
Musical          256
Sport            210
Western          101
War               61
Reality-TV        31
Talk-Show         13
News              10
Adult              6
Game-Show          1
Short              1
Name: count, dtype: int64

### Merging the Datasets

First, I merged the movie information data and rating information data into one dataset called movie and rating so that I could collect all the information on a movie before merging the resulting dataset with the box office data.

I then removed the unnecessary columns e.g start_year as we already have the year column in the box office information data.

In [151]:
movie_and_rating = movie_info.merge(ratings_data, on='tconst', how='inner')
movie_and_rating

tconst                    primary_title              original_title  \
0      tt0063540                        Sunghursh                   Sunghursh   
1      tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2      tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3      tt0100275         The Wandering Soap Opera       La Telenovela Errante   
4      tt0137204                  Joe Finds Grace             Joe Finds Grace   
...          ...                              ...                         ...   
65715  tt9911774        Padmavyuhathile Abhimanyu   Padmavyuhathile Abhimanyu   
65716  tt9913056                     Swarm Season                Swarm Season   
65717  tt9913084                 Diabolik sono io            Diabolik sono io   
65718  tt9914286                Sokagin Çocuklari           Sokagin Çocuklari   
65719  tt9916160                       Drømmeland                  Drømmeland   

       start_year  runtime_minutes                      genres  Main_genres  \
0            2013            175.0          Action,Crime,Drama       Action   
1            2019            114.0             Biography,Drama    Biography   
2            2018            122.0                       Drama        Drama   
3            2017             80.0        Comedy,Drama,Fantasy       Comedy   
4            2017             83.0  Adventure,Animation,Comedy    Adventure   
...           ...              ...                         ...          ...   
65715        2019            130.0                       Drama        Drama   
65716        2019             86.0                 Documentary  Documentary   
65717        2019             75.0                 Documentary  Documentary   
65718        2019             98.0                Drama,Family        Drama   
65719        2019             72.0                 Documentary  Documentary   

       averagerating  numvotes  
0                7.0        77  
1                7.2        43  
2                6.9      4517  
3                6.5       119  
4                8.1       263  
...              ...       ...  
65715            8.4       365  
65716            6.2         5  
65717            6.2         6  
65718            8.7       136  
65719            6.5        11  

[65720 rows x 9 columns]

In [157]:
combined_film_data = box_office_data.merge(
    movie_and_rating, left_on='title', right_on='primary_title', how= 'inner'
)
combined_film_data = combined_film_data.drop(columns=['tconst', 'primary_title', 'original_title', 'start_year', 'genres'])
combined_film_data

title      studio  domestic_gross  \
0                                 Toy Story 3          BV     415000000.0   
1                                   Inception          WB     292600000.0   
2                         Shrek Forever After        P/DW     238700000.0   
3                  The Twilight Saga: Eclipse        Sum.     300500000.0   
4                                  Iron Man 2        Par.     312400000.0   
...                                       ...         ...             ...   
1762               Bilal: A New Breed of Hero          VE        491000.0   
1763                          I Still See You         LGF          1400.0   
1764                    The Catcher Was a Spy         IFC        725000.0   
1765                               Time Freak  Grindstone         10000.0   
1766  Antonio Lopez 1970: Sex Fashion & Disco          FM         43200.0   

      foreign_gross  year  runtime_minutes Main_genres  averagerating  \
0       652000000.0  2010            103.0   Adventure            8.3   
1       535700000.0  2010            148.0      Action            8.8   
2       513900000.0  2010             93.0   Adventure            6.3   
3       398000000.0  2010            124.0   Adventure            5.0   
4       311500000.0  2010            124.0      Action            7.0   
...             ...   ...              ...         ...            ...   
1762      1700000.0  2018            105.0      Action            8.0   
1763      1500000.0  2018             98.0     Fantasy            5.7   
1764       229000.0  2018             98.0   Biography            6.2   
1765       256000.0  2018            104.0      Comedy            5.7   
1766        30000.0  2018             95.0   Biography            6.5   

      numvotes  
0       682218  
1      1841066  
2       167532  
3       211733  
4       657690  
...        ...  
1762     16854  
1763      5010  
1764      4653  
1765      3455  
1766       102  

[1767 rows x 9 columns]

Combining the three files yields a single dataset for feature engineering and analysis. I exclude any unmatched data to ensure there are no missing values for some features.

I cleaned the column names to bring about consistency in the column names

In [158]:
# Creating a dictionary with cleaned column names
cleaned_cols = {
  'title':'Title',
  'studio':'Studio',
  'domestic_gross':'Domestic_Gross',
  'foreign_gross':'Foreign_Gross',
  'Main_genres':'Main_Genres',
  'year':'Year',
  'runtime_minutes':'Runtime_Minutes',
  'averagerating':'Average_Rating',
  'numvotes':'Num_Votes'
}

# Modifying the column names to the cleaned ones
combined_film_data.columns = combined_film_data.columns.map(cleaned_cols)
combined_film_data.columns

Index(['Title', 'Studio', 'Domestic_Gross', 'Foreign_Gross', 'Year',
       'Runtime_Minutes', 'Main_Genres', 'Average_Rating', 'Num_Votes'],
      dtype='object')

In [159]:
combined_film_data.head()

Title Studio  Domestic_Gross  Foreign_Gross  Year  \
0                 Toy Story 3     BV     415000000.0    652000000.0  2010   
1                   Inception     WB     292600000.0    535700000.0  2010   
2         Shrek Forever After   P/DW     238700000.0    513900000.0  2010   
3  The Twilight Saga: Eclipse   Sum.     300500000.0    398000000.0  2010   
4                  Iron Man 2   Par.     312400000.0    311500000.0  2010   

   Runtime_Minutes Main_Genres  Average_Rating  Num_Votes  
0            103.0   Adventure             8.3     682218  
1            148.0      Action             8.8    1841066  
2             93.0   Adventure             6.3     167532  
3            124.0   Adventure             5.0     211733  
4            124.0      Action             7.0     657690

### Feature Engineering

I created a Total gross column which is the sum of both domestic and international profits made by the film.



In [160]:
combined_film_data['Total_gross'] = (
    combined_film_data['Domestic_Gross'] + combined_film_data['Foreign_Gross']
)
combined_film_data.head()

Title Studio  Domestic_Gross  Foreign_Gross  Year  \
0                 Toy Story 3     BV     415000000.0    652000000.0  2010   
1                   Inception     WB     292600000.0    535700000.0  2010   
2         Shrek Forever After   P/DW     238700000.0    513900000.0  2010   
3  The Twilight Saga: Eclipse   Sum.     300500000.0    398000000.0  2010   
4                  Iron Man 2   Par.     312400000.0    311500000.0  2010   

   Runtime_Minutes Main_Genres  Average_Rating  Num_Votes   Total_gross  
0            103.0   Adventure             8.3     682218  1.067000e+09  
1            148.0      Action             8.8    1841066  8.283000e+08  
2             93.0   Adventure             6.3     167532  7.526000e+08  
3            124.0   Adventure             5.0     211733  6.985000e+08  
4            124.0      Action             7.0     657690  6.239000e+08

In [162]:
combined_film_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1767 entries, 0 to 1766
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Title            1767 non-null   object 
 1   Studio           1767 non-null   object 
 2   Domestic_Gross   1767 non-null   float64
 3   Foreign_Gross    1767 non-null   float64
 4   Year             1767 non-null   int64  
 5   Runtime_Minutes  1767 non-null   float64
 6   Main_Genres      1767 non-null   object 
 7   Average_Rating   1767 non-null   float64
 8   Num_Votes        1767 non-null   int64  
 9   Total_gross      1767 non-null   float64
dtypes: float64(5), int64(2), object(3)
memory usage: 138.2+ KB


The resulting dataset has 1767 rows and 9 columns. The datatypes have been converting to the required one and there are no missing values in the dataset.

I saved the resulting dataframe to a csv file named Cleaned_Dataset.csv, which is accessible in the data folder in this repository.

In [161]:
combined_film_data.to_csv('Cleaned_Dataset.csv', index=False)